WIP: Feats 2020

AelarTheElfRogue opened the pull request:
First pass at coding all of the feats. 

To be done: Eldritch Adept, Metamagic Adept, Fighting Initiate

### Checklist
- [x] Correctly formatted on Printer-Friendly sheet
- [x] Correctly formatted on Colourful sheet

AelarTheElfRogue committed:
Update ua_20200713_Feats-2020.js

Smashman commented on Content/ua_20200713_Feats-2020.js:
```diff
@@ -1,4 +1,4 @@
-var iFileName = "ua_20200713_Feats-2020.js";
+var iFileName = "ua_20200713_Feats_2020.js";
```
This should still be `Feats-2020`.

AelarTheElfRogue replied:
Will make this change

Smashman commented on Content/ua_20200713_Feats-2020.js:
```diff
@@ -8,4 +8,306 @@ SourceList["UA:F20"] = {
 	date : "2020/07/13"
+};
+
+FeatsList["magic initiate [artificer]"] = {
```
This feat is named Artificer Initiate in the source.

AelarTheElfRogue replied:
In the main document, the Magic Initiate feats are named in this manner to keep them grouped together. I figured I would stick with overall convention as to avoid confusion when someone is looking for the feat.

Smashman replied:
Those are variants of the 'Magic Initiate' feat and the class name only shows when you choose a variant.

I think it's best to stick with the feat as named in the source document.

AelarTheElfRogue replied:
Change made

Smashman commented on Content/ua_20200713_Feats-2020.js:
```diff
@@ -8,4 +8,306 @@ SourceList["UA:F20"] = {
+	spellcastingBonus : [{
+		name : "Artificer cantrip",
+		spellcastingAbility : 4,
+		'class' : 'artificer',
```
`class` can be written without the single quotes.

AelarTheElfRogue replied:
The other Magic Initiate feats are written this way. I am more than willing to change it, just want to make sure before breaking from what the current convention is.

Smashman replied:
Hmm. I don't think it's necessary. It might be because class is a reserved word in JS, but I think it ought to work just fine in an object as is. I'd remove the quotes to keep it consistent with the rest of the properties.

Smashman commented on Content/ua_20200713_Feats-2020.js:
```diff
@@ -8,4 +8,306 @@ SourceList["UA:F20"] = {
+	source : [["UA:F20", 1]],
+	description : "I learn one Eldritch Invocation option of my choice from the Warlock class. If it has a prerequisite, I can only take it if I am a warlock and meet the prerequisite. I can replace the invocation whenever I gain a level.",
+    prerequisite : "Spellcasting or Pact Magic feature",
+	prereqeval : "What('Class Features')).toLowerCase().indexOf('spellcasting') === -1) || What('Class Features')).toLowerCase().indexOf('pact magic') === -1)"
```
Please use a function, rather than a string that is `eval`ed.

AelarTheElfRogue replied:
This was mostly a guess as to the correct way to identify whether someone has the Spellcasting or Pact Magic features. Could you provide guidance here?

Smashman replied:
Hmm, will consider. I think this is probably fine, just as a function rather than a string. Will investigate.

Smashman commented on Content/ua_20200713_Feats-2020.js:
```diff
@@ -8,4 +8,306 @@ SourceList["UA:F20"] = {
+		times : 2
+	}, {
+		name : "Artificer 1st-level spell",
+		'class' : 'artificer',
```
`class` can be written without the single quotes.

AelarTheElfRogue replied:
The other Magic Initiate feats are written this way. I am more than willing to change it, just want to make sure before breaking from what the current convention is.

Smashman commented on Content/ua_20200713_Feats-2020.js:
```diff
@@ -8,4 +8,306 @@ SourceList["UA:F20"] = {
+		atkAdd : [
+			function (fields, v) {
+				if (isRangedWeapon && ((/firearm/i).test(theWea.type) || (/firearm/i).test(theWea.list)) && fields.Proficiency) {
+					fields.Description = fields.Description.replace(/(,? ?loading|loading,? ?)/i, '');
```
👍

Smashman commented on Content/ua_20200713_Feats-2020.js:
```diff
@@ -8,4 +8,306 @@ SourceList["UA:F20"] = {
+	source : [["UA:F20", 1]],
+    description : "I learn two metamagic options from the sorcerer class. I can use only one option on a spell unless the option says otherwise. I gain 2 sorcery points, and regain all expended points on a long rest.",
+    prerequisite : "Spellcasting or Pact Magic feature",
+	prereqeval : "What('Class Features')).toLowerCase().indexOf('spellcasting') === -1) || What('Class Features')).toLowerCase().indexOf('pact magic') === -1)",
```
Please use a function, rather than a string that is `eval`ed.

Smashman reviewed (changes requested):
There appear to be a number of indentation issues in the file. Please ensure indentation is consistent within objects.

Further comments below.

AelarTheElfRogue committed:
Update ua_20200713_Feats-2020.js

AelarTheElfRogue committed:
Update ua_20200713_Feats-2020.js

Smashman commented on Content/ua_20200713_Feats-2020.js:
```diff
@@ -3,9 +3,370 @@ RequiredSheetVersion(13);
+	description: "I learn Misty Step and one 1st level spell of my choice from the divination or enchantment school of magic. I can cast each once per lost rest without expending a spell slot, and can cast them if I have the slot to do so.",
+	choices: ["Intelligence", "Wisdom", "Charisma"],
+	"intelligence": {
+		description: "I learn Misty Step and one 1st level spell of my choice from the divination or enchantment school of magic. I can cast each once per lost rest without expending a spell slot, and can cast them if I have the slot to do so. Intelligence is my spellcasting ability [+1 Intelligence]",
```
Perhaps "Charisma is my spellcasting ability for these spells" to make it clear it doesn't wholly affect the class. (For other abilities and for the Shadowfell one too)

AelarTheElfRogue replied:
This actually was an issue fitting it on the Colorful sheet, as there is less room because of the non-monospaced font. I have foudn a way to update the spell description to be less wordy and fit that text.

AelarTheElfRogue committed:
Update ua_20200713_Feats-2020.js

Updated the Fey Touched and Shadow Touched descriptions to fit the phrase "for these spells" to the end on the colorful sheets.

AelarTheElfRogue committed:
Update ua_20200713_Feats-2020.js

Added full descriptions and added prerequisite for Fighting Initiate

AelarTheElfRogue committed:
Update ua_20200713_Feats-2020.js

Updated source brackets

AelarTheElfRogue committed:
Update ua_20200713_Feats-2020.js

Added prereq eval for Fighting Initiate

AelarTheElfRogue committed:
Update ua_20200326_Spells-and-Magic-Tattoos.js

Initial item listing

AelarTheElfRogue committed:
Update ua_20200326_Spells-and-Magic-Tattoos.js

AelarTheElfRogue committed:
Update ua_20200713_Feats-2020.js

AelarTheElfRogue committed:
Update ua_20200713_Feats-2020.js

Capitalized the P in Piercer feat name
